api/__init__.py

```diff
@@ -266,6 +266,12 @@ def split_appname_appversion(long_app_id):
         appname = appname[:-len(appversion) - 1]
     if appname.endswith('/'):
         appname = appname[:-1]
+    if appname == 'apps.clams.ai':
+        # NOTE (mv @ 4/18/25): this is to deal with the case where the app value does
+        # not have a version at all (for example http://apps.clams.ai/tesseract), in
+        # which case the appname would be set to apps.clams.ai. This was possibly only
+        # needed because I used some older data.
+        appversion = None
```

Comment:
In this case, shouldn't we actually properly handle to set `appname==tesseract` and `appversion==None`, instead of leaving this as future debt, until someone else re-discover the same problem when they try to push MMIFs from `http://apps.clams.ai/tesseract` and `http://apps.clams.ai/whisper` and end up with both MMIFs are stuffed into `/storage/app.clams.ai/...` ?

Reply:
Yeah, probably. I was thinking of reimplementing it using Path.parts and take it from there, which seems more versatile than a right split. Will do this soon.